### PR TITLE
GDB-4807 Do not delete predicate when deleting the object

### PR DESCRIPTION
**Pull request by teodossidossev**

### CODE REVIEW

File: cypress/test/delete.spec.ts [hunk 1]
```diff
@@ -250,6 +250,62 @@ describe('Delete', () => {
...
+      // Then I'll be able to add new object
+      MappingSteps.getTripleObjectValue(1).type('222').blur();
+
+      // Then objects keep order
+      MappingSteps.getTripleObjectValuePreview(0).should('contain', '111');
+      MappingSteps.getTripleObjectValuePreview(1).should('contain', '222');
+      MappingSteps.getTripleObjectValuePreview(2).should('contain', '333');
+    });
+
+    it.only('Should be able to insert subject at same place as the deleted object when property predicate', () => {
```
Comment (svilenvelikov) [inline cypress/test/delete.spec.ts:282]: `only`

File: src/app/services/model-management.service.ts [hunk 2]
```diff
@@ -365,32 +365,48 @@ export class ModelManagementService {
   }
 
   public setTypeMapping(subject: MappingBase, selected: SimpleIRIValueMappingImpl) {
+    let index: number;
```
Comment (svilenvelikov) [inline src/app/services/model-management.service.ts:368]: Eventually rename this `index` something more meaningful

File: src/app/services/model-management.service.ts [hunk 3]
```diff
@@ -365,32 +365,48 @@ export class ModelManagementService {
...
+
+    if (index > -1) {
+      subject.getTypeMappings().splice(index, 1, selected);
+    } else {
+      subject.getTypeMappings().push(selected);
     }
-    subject.getTypeMappings().push(selected);
+  }
+
+  private getEmptyTypeMappingIndex(subject: MappingBase): number {
```
Comment (svilenvelikov) [inline src/app/services/model-management.service.ts:382]: Isn't it better to move these `getEmptyTypeMappingIndex` and `getEmptyValueIndex` functions in the model management service?

File: src/app/services/model-management.service.ts (hunk 2 above)
Comment (teodossidossev) [reply src/app/services/model-management.service.ts:368]: +1

File: src/app/services/model-management.service.ts (hunk 3 above)
Comment (teodossidossev) [reply src/app/services/model-management.service.ts:382]: This is the model management service

File: cypress/test/delete.spec.ts [hunk 4]
```diff
@@ -250,6 +250,62 @@ describe('Delete', () => {
       MappingSteps.getTriples().should('have.length', 2);
       MappingSteps.getTripleObjectValue(1).should('be.empty');
     });
+
+    it('Should be able to insert subject at same place as the deleted object when type predicate', () => {
+      cy.route('GET', '/orefine/command/core/get-models/?project=123', 'fixture:empty-mapping-model.json');
+      cy.route('POST', '/repositories/Movies', 'fixture:edit-mapping/autocomplete-response.json');
```
Comment (svilenvelikov) [inline cypress/test/delete.spec.ts:256]: Please use the fixture from the respective package. This is the delete.spec and not the edit.spec

File: src/app/services/model-management.service.ts (hunk 3 above)
Comment (svilenvelikov) [reply src/app/services/model-management.service.ts:382]: OMG

**Pull request merged**